applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_linear_3D2N.cpp

```diff
@@ -297,6 +310,114 @@ void CrBeamElementLinear3D2N::CalculateOnIntegrationPoints(
     KRATOS_CATCH("");
 }
 
+
+void CrBeamElementLinear3D2N::CalculateRigidityReductionMatrix(
+    BoundedMatrix<double, msElementSize, msElementSize>& rRigidityReductionMatrix) const
+{
+    KRATOS_TRY
+
+
+    rRigidityReductionMatrix = ZeroMatrix(msElementSize, msElementSize);
+    const double E = GetProperties()[YOUNG_MODULUS];
+    const double L = StructuralMechanicsElementUtilities::CalculateReferenceLength3D2N(*this);
+
+
+    const double Iy = GetProperties()[I22];
+    const double Iz = GetProperties()[I33];
+
+    const GeometryType& r_geometry = GetGeometry();
+    
+    // inititalise fixity factors as completely fixed
+    double alpha1_yy = 1;
+    double alpha2_yy = 1;
+
+    double alpha1_zz = 1;
+    double alpha2_zz = 1;
+
+    // set fixity factors, 0 is no fixity, i.e. a hinge; 1 is completely fixed, the resulting stiffness matrix will remain unchanged
+    if (r_geometry[0].Has(ROTATIONAL_STIFFNESS_AXIS_2)) {
+        const double rotational_stiffness_y_1 = r_geometry[0].GetValue(ROTATIONAL_STIFFNESS_AXIS_2);
+        alpha1_zz = (rotational_stiffness_y_1 > 0) ? 1 / (1 + 3 * E * Iy / (rotational_stiffness_y_1 * L)) : 0.0;
+    }

```

Comment:
you can skip the `Has` check. If you make sure that they have a valid input, and the `Geometry` from which you get the nodes from is `const`, then you can simplify to
`const double alpha1_zz = 1 / (1 + 3 * E * Iy / (r_geometry[0].GetValue(ROTATIONAL_STIFFNESS_AXIS_2) * L))`

Reply:
In order to not have a change of behaviour of the existing beam, the default beam should be without hinge. This means that the default ROTATIONAL_STIFFNESS_AXIS_2 should be infinite. In my oppinion it is cleaner to do a check if the ROTATIONAL_STIFFNESS_AXIS_2 is actually defined, thus keeping the Has.